src/sync/sync_manager.hpp

```diff
@@ -66,7 +66,8 @@ friend class SyncSession;
     // Configure the metadata and file management subsystems. This MUST be called upon startup.
     void configure_file_system(const std::string& base_file_path,
                                MetadataMode metadata_mode=MetadataMode::Encryption,
-                               util::Optional<std::vector<char>> custom_encryption_key=none);
+                               util::Optional<std::vector<char>> custom_encryption_key=none,
+                               bool reset_metadata_on_error=false);
```

Comment:
Right now I don't think any bindings need this behavior at least until it needs to be informed that there was an error reading metadata. In this case it's possible to catch an exception,  perform other actions (like reseting user realms) and reset metadata by calling this method again with `reset_metadata_on_error=true`.